### ISSUE TITLE
Player disconnect on Login is handled uncorretclly

### ISSUE THREAD

**Issue by Xayanix** [ProtocolSupport/ProtocolSupport#369]
Issue: https://github.com/ProtocolSupport/ProtocolSupport/issues/349 is still active.
I thought that updating bungee fixed problem, but it didn't.

Disallowing AsyncPlayerPreLoginEvent is handled correctly, but PlayerLoginEvent not.

**Comment by Speentie8081** [comment]
Same thing happens to me, very interesting, I just use PlayerPreLoginEvent and handle everything there.

**Comment by Xayanix** [comment]
Would you fix it, please?

**Comment by Shevchik** [comment]
That issue is at low priority. So it will take time before i even start working at this.

**Comment by Xayanix** [comment]
I noticed that when player got kicked from server, then he does immune every velocity effects.

**Comment by Xayanix** [comment]
It is really important for me.

**Comment by Xayanix** [comment]
bump

**Comment by Rida94** [comment]
@Shevchik anything new about this?

**Comment by Shevchik** [comment]
No. That requires an additional plugin bungee side, so this fix will be added to ProtocolSupportBungee, but it's not ready yet

**Comment by 0XE4** [comment]
We are getting this on servers that do not use Bungeecord as well, on the current version of Paper. Console is showing 'PlayerName was kicked due to keepalive timeout!' as soon as they log in before kicking them.

**Comment by Shevchik** [comment]
Your issue is a completely separate issue caused by rogue keep alive packets sent by something on your server.
